### PR TITLE
Allow mining min diff for very slow (2h+) blocks

### PR DESCRIPTION
**Pull request by UdjinM6**
Add additional rule for some extreme cases. Testnet/devnet only ofc.

### CODE REVIEW

File: src/pow.cpp [hunk 1]
```diff
@@ -109,13 +109,18 @@ unsigned int static DarkGravityWave(const CBlockIndex* pindexLast, const CBlockH
         if (pindexLast->nChainWork >= UintToArith256(uint256S("0x000000000000000000000000000000000000000000000000003ff00000000000"))
             // and immediately on devnet
             || !params.hashDevnetGenesisBlock.IsNull()) {
+            // recent block is more than 10 minues old
```
Comment (nmarley) [inline src/pow.cpp]: nit: minutes

Comment (UdjinM6) [reply src/pow.cpp]: 🙈

File: src/pow.cpp [hunk 2]
```diff
@@ -109,13 +109,18 @@ unsigned int static DarkGravityWave(const CBlockIndex* pindexLast, const CBlockH
...
             || !params.hashDevnetGenesisBlock.IsNull()) {
+            // recent block is more than 10 minutes old
             if (pblock->GetBlockTime() > pindexLast->GetBlockTime() + params.nPowTargetSpacing*4) {
                 arith_uint256 bnNew = arith_uint256().SetCompact(pindexLast->nBits) * 10;
                 if (bnNew > bnPowLimit) {
                     bnNew = bnPowLimit;
                 }
                 return bnNew.GetCompact();
             }
+            // recent block is more than 2 hours old
```
Comment (nmarley) [inline src/pow.cpp]: Will it ever get here, or would the condition above catch it first?

> pblock->GetBlockTime() > pindexLast->GetBlockTime() + params.nPowTargetSpacing*4

(Should this block be moved up above?)

Comment (nmarley) [reply src/pow.cpp]: @UdjinM6 nvm, I see you caught it too

Comment (UdjinM6) [reply src/pow.cpp]: yep, 28be742 :)

**Pull request merged**